sdk/digitaltwins/Azure.DigitalTwins.Core/src/Queries/QueryBuilders/SelectQuery.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Globalization;
+using Azure.DigitalTwins.Core.Queries.QueryBuilders;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Azure.DigitalTwins.Core.QueryBuilder
+{
+    // NOTE -- accessibility internal for now to bypass needing static.
+    /// <summary>
+    /// Custom ADT query builder class that facilitates building queries against an ADT instance.
+    /// </summary>
+    public sealed class SelectQuery : QueryBase
+    {
+        private readonly FromQuery _delegationClause;
+        private readonly AdtQueryBuilder _parent;
+        private SelectClause _clause;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SelectQuery"/> class.
```

Comment:
nice, you can use this format for all your class constructors. especially if they are public